containers/ddev-router/Dockerfile

```diff
@@ -6,22 +6,60 @@ ENV DEBIAN_FRONTEND noninteractive
 ENV DOCKER_GEN_VERSION 0.7.4
 ENV DOCKER_HOST unix:///tmp/docker.sock
 
+# TARGETPLATFORM is Docker buildx's target platform (e.g. linux/arm64), while 
+# BUILDPLATFORM is the platform of the build host (e.g. linux/amd64)
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+
+SHELL ["/bin/bash", "-c"]
+
+RUN mkdir /tmp/ddev && \
+    if [[ $TARGETPLATFORM == "linux/arm64" ]]; then \
+        printf '# jwilder/forego doesnt have an arm64 download so we resort to the original forego repo \n\
+FORGO_IS_TGZ="true" \n\
+FORGO_FILENAME="forego-stable-linux-arm64.tgz" \n\
+FORGO_URL="https://bin.equinox.io/c/ekMN3bCZFUn/forego-stable-linux-arm64.tgz" \n\
+# jwilder/docker-gen doesnt have an arm64 download so we created a custom build \n\
+DOCKER_GEN_URL="https://github.com/dennisameling/docker-gen/releases/download/0.7.5/docker-gen-linux-arm64-0.7.5.tar.gz" \n\
+DOCKER_GEN_FILE="docker-gen-linux-arm64-0.7.5.tar.gz" \n\
+# PR for official arm64 support pending: https://github.com/FiloSottile/mkcert/pull/284 \n\
+MKCERT_URL="https://github.com/dennisameling/mkcert/releases/download/${MKCERT_VERSION}-ddev/mkcert-${MKCERT_VERSION}-linux-arm64"' > /tmp/ddev/vars; \
+
+    else \
+        printf 'FORGO_IS_TGZ="false" \n\
+FORGO_URL="https://github.com/jwilder/forego/releases/download/v0.16.1/forego" \n\
+DOCKER_GEN_URL="https://github.com/jwilder/docker-gen/releases/download/${DOCKER_GEN_VERSION}/docker-gen-linux-amd64-${DOCKER_GEN_VERSION}.tar.gz" \n\
+DOCKER_GEN_FILE="docker-gen-linux-amd64-${DOCKER_GEN_VERSION}.tar.gz" \n\
+MKCERT_URL="https://github.com/FiloSottile/mkcert/releases/download/$MKCERT_VERSION/mkcert-${MKCERT_VERSION}-linux-amd64"' > /tmp/ddev/vars; \
+    fi
```

Comment:
Hope the approach above is OK for multi-arch support! Was the only proper way I could find to share variables cross-build steps